avalanche/distributed/distributed_batch.py

```diff
@@ -0,0 +1,129 @@
+from abc import abstractmethod, ABC
+from typing import TypeVar, List, Optional, Callable, Any, Tuple
+
+import torch
+from torch import Tensor
+
+from avalanche.distributed import DistributedHelper
+from avalanche.distributed.distributed_value import SwitchableDistributedValue
+
+LocalT = TypeVar('LocalT')
+DistributedT = TypeVar('DistributedT')
+
+
+class DistributedObject(SwitchableDistributedValue[LocalT, DistributedT], ABC):
+    """
+    An intermediate abstract class in charge of synchronizing objects.
+
+    The merge procedure must be implemented in child classes.
+    """
+    def _synchronize_distributed_value(self) -> DistributedT:
+        objects = self._synchronize_objects()
+        return self._merge_objects(objects)
+
+    def _synchronize_objects(self) -> List[LocalT]:
+        return DistributedHelper.gather_all_objects(
+            self._local_value
+        )
+
+    @abstractmethod
+    def _merge_objects(self, objects: List[LocalT]) -> DistributedT:
+        pass
+
+
+class DistributedBatch(DistributedObject[LocalT, LocalT], ABC):
+    """
+    An intermediate abstract class in charge of synchronizing data batches.
+
+    This class can handle batches as either tuples of elements (as usual) or
+    even single values.
+
+    The merge procedure of tuples and single elements must be implemented in
+    child classes. By default, the tuples will be merged value by value.
+
+    NOTE: In the future, this class may be replaced with a version in which only
+    the accessed tuple elements are synchronized, instead of the whole batch.
+    The current design, in which child classes have to implement
+    `_merge_single_values`, allows for this change to happen without affecting
+    child classes.
+    """
+
+    def __init__(self, name: str, initial_local_value: LocalT):
+        super().__init__(name, initial_local_value)
+        self._value_is_tuple = False
+
+    def _synchronize_distributed_value(self) -> LocalT:
```

Comment:
can we abbreviate the names? `_set_local`, `_synchronize`, `_merge`

Reply:
I think that `_synchronize` makes sense.